featuretools/synthesis/dfs.py

```diff
@@ -261,6 +262,10 @@ def dfs(dataframes=None,
     features = dfs_object.build_features(
         verbose=verbose, return_types=return_types)
 
+    assert (isinstance(features, list) and features != [] and
+            all([isinstance(feature, FeatureBase) for feature in features])), \
+        "No features can be generated from the specified primitives. Please make sure the primitives you are using are compatible with the variable types in your data."
+
```

Comment:
Let's move this check below where a `features_only` run of `dfs` would return so this check does not cause a `features_only` run of `dfs` to error if no features were created